src/rebuild/scan.c

```diff
@@ -633,29 +647,96 @@ placement_check(uuid_t co_uuid, vos_iter_entry_t *ent, void *data)
 	if (map != NULL)
 		pl_map_decref(map);
 
+	if (--xarg->yield_freq == 0) {
+		xarg->yield_freq = DEFAULT_YIELD_FREQ;
+		ABT_thread_yield();
+		*acts |= VOS_ITER_CB_YIELD;
+	}
+
 	return rc;
 }
 
-struct rebuild_iter_arg {
-	ds_iter_cb_t	callback;
-	void		*arg;
-};
+static int
+rebuild_container_scan_cb(daos_handle_t ih, vos_iter_entry_t *entry,
+			  vos_iter_type_t type, vos_iter_param_t *iter_param,
+			  void *data, unsigned *acts)
```

Comment:
This 'acts' isn't used, so yield in this callback will break the vos_iterate(VOS_ITER_COUUID), so we need to either:
- Mark yield in 'acts' once following dtx_resync() or vos_iterate() yield, that require more code changes to return back yield information from these two functions; or
- Always mark yield in 'acts', so that vos_iterate(VOS_ITER_COUUID) will re-probe on every entry. (assuming the quantity of container is small, the extra re-probe isn't that bad)

In either way, I think the last processed container needs be tracked in the scan argument to avoid re-processing it again.

Reply:
I may not follow here.
rebuild object scanning yield happens in rebuild_obj_scan_cb, instead of container iteration.
If I do not touch acts here, then it will not yield for this iteration?